server/protocols/jmap-integration-testing/cassandra-jmap-integration-testing/src/test/java/org/apache/james/jmap/cassandra/CassandraSetMessagesMethodWithBigMessageTestExtension.java

```diff
@@ -0,0 +1,87 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.cassandra;
+
+import static org.apache.james.CassandraJamesServerMain.ALL_BUT_JMX_CASSANDRA_MODULE;
+
+import org.apache.james.DockerCassandraTestExtension;
+import org.apache.james.DockerElasticSearchTestExtension;
+import org.apache.james.GuiceJamesServer;
+import org.apache.james.jmap.methods.integration.SetMessagesMethodWithBigMessageTestSystem;
+import org.apache.james.modules.TestJMAPServerModule;
+import org.apache.james.server.core.configuration.Configuration;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+import org.junit.rules.TemporaryFolder;
+
+public class CassandraSetMessagesMethodWithBigMessageTestExtension implements BeforeEachCallback, AfterEachCallback, ParameterResolver {
+
+    private static final int LIMIT_TO_10_MESSAGES = 10;
+
+    private SetMessagesMethodWithBigMessageTestSystem testSystem;
+    private TemporaryFolder temporaryFolder;
+
+    @Override
+    public void beforeEach(ExtensionContext extensionContext) throws Exception {
+        temporaryFolder = new TemporaryFolder();
+        temporaryFolder.create();
+
+        ExtensionContext.Store store = extensionContext.getStore(ExtensionContext.Namespace.GLOBAL);
```

Comment:
how are you supposed to know that this CassandraSetMessagesMethodWithBigMessageTestExtension depends on some other extensions ? This extension gives a false feeling of independence.
BTW, the global store thing is suspicious to me

Reply:
> how are you supposed to know that this CassandraSetMessagesMethodWithBigMessageTestExtension depends on some other extensions ?
I don't understand

`CassandraSetMessagesMethodWithBigMessageTestExtension` depends on others extensions, yes, i want that, this extension need `DockerCassandraTestExtension` & `DockerElasticSearchTestExtension` to get their information and start james server. I want to leverage the reusable of `DockerCassandraTestExtension` & `DockerElasticSearchTestExtension` for other extensions that need docker cassandra or docker elasticsearch instead of duplicating starting docker cassandra & docker elasticsearch every where.

> BTW, the global store thing is suspicious to me
you can imagine `GLOBAL` Namespace is just a key object, we get data from a map by a key object. I can create another Namespace to share data between extensions but I prefer an existing one.

Reply:
> I don't understand

The fact is CassandraSetMessagesMethodWithBigMessageTestExtension depends on DockerCassandraTestExtension and DockerElasticSearchTestExtension to run.

Another fact is : CassandraSetMessagesMethodWithBigMessageTestExtension doesn't have an explicit reference to these two extensions

You are passing dependencies by a global map and hide this dependency link, thus a user of CassandraSetMessagesMethodWithBigMessageTestExtension will discover the requirement at run time instead of build time.

You should have a direct (build time) dependency between these classes.